run_action.py

```diff
@@ -755,16 +771,38 @@ def main():
             args.pull_request_id,
             warning_comment_prefix=warning_comment_prefix,
             auto_resolve_conversations=args.auto_resolve_conversations == "true",
-            single_comment_marker=single_comment_marker,
+            single_comment_markers=single_comment_markers,
         )
         return 0
 
+    clang_tidy_errors = [
+        fix for fix in clang_tidy_fixes["Diagnostics"] if fix["Level"] == "Error"
+    ]
+    clang_tidy_warnings = [
+        fix for fix in clang_tidy_fixes["Diagnostics"] if fix["Level"] == "Warning"
+    ]
+    clang_tidy_remarks = [
+        fix for fix in clang_tidy_fixes["Diagnostics"] if fix["Level"] == "Remark"
+    ]
+
+    if len(clang_tidy_fixes["Diagnostics"]) != (
+        len(clang_tidy_errors) + len(clang_tidy_warnings) + len(clang_tidy_remarks)
+    ):
+        print(
+            "WARNING: some fixes have an unexpected Level (e.g. not Error, Warning, Remark)"
+        )
+
+    # show diagnostics ordered by level: first error, then warning, then remark
+    clang_tidy_fixes["Diagnostics"] = (
+        clang_tidy_errors + clang_tidy_warnings + clang_tidy_remarks
+    )
+
```

Comment:
Hi @fwc as far as I understand this logic, if some other level is added in the future, other than `Error`, `Warning` or `Remark`, then it will not be processed (even as a fallback), right? There will be a warning in the console, but the corresponding diagnostics will not be shown?

Reply:
> There will be a warning in the console, but the corresponding diagnostics will not be shown?

Yes, good point.

I addressed this, now no diagnostics should be lost.

Reply:
Looks good to me, thank you!